### PR TITLE
UIF-519: Replace deprecated calls to transaction API

### PR DESCRIPTION
**Pull request by alisher-epam**
## Purpose

[UIF-519](https://folio-org.atlassian.net/browse/UIF-519): Replace deprecated calls to transaction API
## Approach

https://github.com/user-attachments/assets/764fa9d6-040d-4b05-88c3-2dcb2bee4cf3

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by SerhiiNosko — APPROVED** [review]

File: src/Transactions/CreateTransaction/CreateTransactionContainer.js [hunk 1]
```diff
@@ -121,39 +89,76 @@ export const CreateTransactionContainer = ({
...
+
+    showCallout({ message, type: 'error' });
+  }, [getAccumulatedDataObject, handleCreateTransactionErrorResponse, showCallout, transactionTypeKey]);
+
+  const saveTransactionStep = useCallback(async (formValues, { resultBudgetName }) => {
+    return batchTransactions({
+      transactionType: BATCH_TRANSACTION_TYPES.transactionsToCreate,
+      data: [{
+        ...omit(formValues, ['fundId']),
+        id: uuidv4(),
```
Comment (usavkov-epam) [inline src/Transactions/CreateTransaction/CreateTransactionContainer.js:113]: Should this be within the UI's responsibility? cc @SerhiiNosko

**Review by usavkov-epam — COMMENTED** [review]
Good job!
There is only one question I want to clarify before approve: setting UUID of the transactions on UI side.

**Comment by usavkov-epam** [comment]
And, please do not ignore the pre-merge checklist and screenshots in the pull request description.

File: src/Transactions/CreateTransaction/CreateTransactionContainer.js (hunk 1 above)
Comment (alisher-epam) [reply src/Transactions/CreateTransaction/CreateTransactionContainer.js:113]: Currently, the BE prevents us from making a request without a unique `id`Screenshot 2024-11-04 at 16 52 16

**Review by usavkov-epam — APPROVED** [review]

**Pull request merged**